src/time.cpp

```diff
@@ -21,21 +57,21 @@ namespace g3 {
       // This is needed since latest version (at time of writing) of gcc4.7 does not implement this library function yet.
       // return value is SIMPLIFIED to only return a std::string
 
-      std::string put_time(const struct tm *tmb, const char *c_time_format) {
+      std::string put_time(const struct tm* tmb, const char* c_time_format) {
 #if (defined(WIN32) || defined(_WIN32) || defined(__WIN32__)) && !defined(__MINGW32__)
          std::ostringstream oss;
          oss.fill('0');
          // BOGUS hack done for VS2012: C++11 non-conformant since it SHOULD take a "const struct tm*  "
-         oss << std::put_time(const_cast<struct tm *> (tmb), c_time_format);
+         oss << std::put_time(const_cast<struct tm*> (tmb), c_time_format);
          return oss.str();
 #else    // LINUX
          const size_t size = 1024;
          char buffer[size]; // IMPORTANT: check now and then for when gcc will implement std::put_time.
          //                    ... also ... This is way more buffer space then we need
 
          auto success = std::strftime(buffer, size, c_time_format, tmb);
-         if (0 == success)
-         {
+         // TODO: incorrect code
```

Comment:
No code change but added a comment so that code readers that have not seen this pattern before will be less baffled by it.    It is essentially a way of exiting with a clear message in debug mode. In production mode a safe value is instead returned.